lib/rucio/daemons/auditor/srmdumps.py

```diff
@@ -234,29 +238,63 @@ def download_rse_dump(rse, configuration, date='latest', destdir=DUMPS_CACHE_DIR
     '''
     logger = logging.getLogger('auditor.srmdumps')
     base_url, url_pattern = generate_url(rse, configuration)
-    if date == 'latest':
-        logger.debug('Looking for site dumps in: "%s"', base_url)
-        links = get_links(base_url)
-        url, date = get_newest(base_url, url_pattern, links)
-    else:
-        url = '{0}/{1}'.format(base_url, date.strftime(url_pattern))
 
     if not os.path.isdir(destdir):
         os.mkdir(destdir)
 
-    filename = '{0}_{1}_{2}_{3}'.format(
-        'ddmendpoint',
-        rse,
-        date.strftime('%d-%m-%Y'),
-        hashlib.sha1(url.encode()).hexdigest()
-    )
-    filename = re.sub(r'\W', '-', filename)
-    path = os.path.join(destdir, filename)
-
-    if not os.path.exists(path):
-        logger.debug('Trying to download: "%s"', url)
-        with temp_file(destdir, final_name=filename) as (f, _):
-            download(url, f)
+    # check for objectstores, which need to be handled differently
+    rse_id = get_rse_id(rse)
+    rse_attr = list_rse_attributes(rse_id)
+    if 'is_object_store' in rse_attr and rse_attr['is_object_store'] is not False:
+        tries = 1
+        if date is None:
+            # on objectstores, can't list dump files, so try the last N dates
+            date = datetime.datetime.now()
+            tries = OBJECTSTORE_NUM_TRIES
+        path = ''
+        while tries > 0:
+            url = '{0}/{1}'.format(base_url, date.strftime(url_pattern))
+
+            filename = '{0}_{1}_{2}_{3}'.format(
+                'ddmendpoint',
+                rse,
+                date.strftime('%d-%m-%Y'),
+                hashlib.sha1(url.encode()).hexdigest()
+            )
+            filename = re.sub(r'\W', '-', filename)
+            path = os.path.join(destdir, filename)
+            if not os.path.exists(path):
+                logger.debug('Trying to download: "%s"', url)
+                if 'sign_url' in rse_attr:
+                    url = get_signed_url(rse_id, rse_attr['sign_url'], 'read', url)
+                try:
```

Comment:
On a second look, another note.

The call to download(url, f) may throw different exceptions, i.e. gfal2.Error in https://github.com/rucio/rucio/blob/0f674128c48eef880eb9362bb818de3fbb6a7154/lib/rucio/common/dumper/__init__.py#L294 

Also, binary=True in L271 won't work if the download is done with gfal 
```
  File "/usr/local/lib/python3.6/site-packages/rucio/common/dumper/__init__.py", line 297, in gfal_download_to_file
    file_.write(chunk)
TypeError: a bytes-like object is required, not 'str'
```